util/check-format-commit.sh

```diff
@@ -45,10 +45,23 @@ cleanup() {
 
 trap cleanup EXIT
 
-# Get the canonical sha256 sum for the commit we are checking
+# Get the canonical sha256 sum for the commits we are checking
 # This lets us pass in symbolic ref names like master/etc and 
 # resolve them to sha256 sums easily
-COMMIT=$(git rev-parse $1)
+COMMIT_RANGE="$@"
+COMMIT_LAST=$(git rev-parse $COMMIT_RANGE)
+
+# If the commit range was just one single revision, git rev-parse
+# will output jut commit id of that one alone.  In that case, we
+# must manipulate a little to get a desirable result, 'cause git
+# diff has a slightly different interpretation of a single commit
+# id, and takes that to mean all commits up to HEAD.
+if [ $(echo "$COMMIT_LAST" | wc -l) -gt 1 ]; then
+    COMMIT_LAST=$(echo "$COMMIT_LAST" | head -1)
+else
+    # $COMMIT_RANGE is just one commit, make it an actual range
+    COMMIT_RANGE=$COMMIT_RANGE^..$COMMIT_RANGE
+fi
 
 # Fail gracefully if git rev-parse doesn't produce a valid
 # commit
```

Comment:
Shouldn't this test really happen immediately after the `git rev-parse` call?

Reply:
Fixing